File: MinElemLinkedList/Find.java
import java.lang.*;
import java.util.*;

class Node 
{
    int data;
    Node next;    

    Node()
    {
        this.data = 0;
        this.next = null;
    }
}

class SinglyLL
{
    Node Head;

    SinglyLL()
    {
        this.Head = null;
    }

    void InsertFirst(int value)
    {
        Node newn = new Node();
        newn.data = value;
        newn.next = null;

        if(this.Head == null)
        {
            this.Head = newn;
        }
        else 
        {
            newn.next = this.Head;
            this.Head = newn;
        }
    }

    void InsertLast(int value)
    {
        Node newn = new Node();
        newn.data = value;
        newn.next = null;
        Node temp = this.Head;

        if(this.Head == null)
        {
            this.Head = newn;
        }
        else
        {
            while(temp.next != null)
            {
                temp = temp.next;
            }
            temp.next = newn;
        }
    }
    void Display()
    {
        Node temp = this.Head;

        while(temp != null)
        {
            System.out.printf("|%d|->",temp.data);
            temp = temp.next;
        }
    }

    int Count()
    {
        Node temp = this.Head;
        int icnt = 0;

        while(temp != null)
        {
            icnt++;
            temp = temp.next;
        }

        return icnt;
    }

    int SmallElem()
    {
        Node temp = this.Head;
        int Min = 0;

        Min = temp.data;
        while(temp != null)
        {
            if(temp.data < Min)
            {
                Min = temp.data;
            }
            temp = temp.next;
        }

        return Min;
    }
}
class Find
{
    public static void main(String args[])
    {
        int iret = 0;

        System.out.println("Jay Ganesh...");

        SinglyLL sobj = new SinglyLL();
        sobj.InsertFirst(51);
        sobj.InsertFirst(11);
        sobj.InsertFirst(10);
        sobj.InsertLast(2);
        sobj.InsertLast(101);
        sobj.InsertLast(105);

        System.out.println("Elements of Linked List are : ");
        sobj.Display();

        iret = sobj.Count();
        System.out.printf("\nCount of Elements From Linked list Is : %d\n",iret);
        
        iret = sobj.SmallElem();
        System.out.printf("Smallest Element From Linked list Is : %d\n",iret);
    }  
}

File: MinElemLinkedList/Find.py
class Node :
    data = 0;
    next = None;

    def __init__(self) : 
        self.data = 0;
        self.next = None;
    
class SinglyLL :
    Head = None;

    def __init__(self) : 
        self.Head = None;
    
    def InsertFirst(self,value) :
        newn = Node();
        newn.data = value;
        newn.next = None;

        if(self.Head == None) :
            self.Head = newn;
        else : 
            newn.next = self.Head;
            self.Head = newn;
 
    def InsertLast(self,value) :
        newn = Node();
        newn.data = value;
        newn.next = None;
        temp = self.Head;

        if(self.Head == None) :
            self.Head = newn;
        else : 
            while(temp.next != None) :
                temp = temp.next;
            temp.next = newn;

    def Display(self) : 
        temp = self.Head;

        while(temp != None) :
            print("|%d|->"%temp.data,end="");
            temp = temp.next;
    
    def Count(self) :
        icnt = 0;
        temp = self.Head;
    
        while(temp != None) :
            icnt = icnt + 1;
            temp = temp.next;

        return icnt;

    def SmallElem(self) :
        Min = 0;
        temp = self.Head;

        Min = temp.data;
        while(temp != None) :
            if(temp.data < Min) :
                Min = temp.data;
            temp = temp.next;

        return Min;

def main() :
    print("Jay ganesh...");
    iret = 0;

    sobj = SinglyLL();
    sobj.InsertFirst(51);
    sobj.InsertFirst(11);
    sobj.InsertFirst(10);
    sobj.InsertLast(2);
    sobj.InsertLast(101);
    sobj.InsertLast(105);

    print("Element From Linked List are : ");
    sobj.Display();

    iret = sobj.Count();
    print("\nCount of Elements From Linked List is :",iret);

    iret = sobj.SmallElem();
    print("Smallest Element From Linked List is :",iret);

if __name__ == "__main__" :
    main();